### DISCUSSION TITLE
How to add tag to circuit via API or nbshell

### DISCUSSION THREAD

**Discussion by el-magneto** [netbox-community/netbox#8721] [Q&A] [answered]
Hi,
I have no idea how to add tag to circuit in my script.
I tried couple of ways but somtimes I got error and somtimes got no error but nothing happens like here:
```
 bac_circuit = Circuit.objects.get(cid=data['isp_backup'])
 bac_circuit.tags= [Tag.objects.get(name="BACKUP")]
 bac_circuit.save()
```
I checked it in nbshell and can see that bac_circuit.tags is an object:

```
>>> bac_circuit = Circuit.objects.get(cid='123')
>>> bac_circuit
<Circuit: 123>
>>> bac_circuit.cid
'123'
>>> bac_circuit.provider
<Provider: Orange>
>>> bac_circuit.tags
<taggit.managers._TaggableManager object at 0x7efe866eb040>
>>> bac_circuit.tags.all()
<RestrictedQuerySet [<Tag: TEST>]>
>>> bac_circuit.tags.all()[0]
<Tag: TEST>
>>>
```
How to add new (already configured tag) to that object?

**Comment by candlerb** [comment, marked as answer]
'tags' is some sort of Django ORM manager object.  Try:

```
bac_circuit.tags.set([Tag.objects.get(name="BACKUP")])
```
or
```
bac_circuit.tags.set(["BACKUP"])
```
I'm reasonably sure the first will work, not sure about the second.

**Reply by el-magneto** [reply]
Thank you! It works in nbshell and API!

**Comment by hagbarddenstore** [comment]
FYI, if you’re adding a single tag, just do tags.add(myTag)
